### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by IgorBochkor**
added race condition

### CODE REVIEW

File: log/logger.log [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+ 16:41:36,159 INFO  MyThread:17 - 33
+ 16:41:36,159 INFO  MyRunnable:17 - 32
+ 16:41:36,159 INFO  MyThread:17 - 34
+ 16:41:36,159 INFO  MyRunnable:17 - 35
+ 16:41:36,159 INFO  MyThread:17 - 36
+ 16:41:36,160 INFO  MyRunnable:17 - 37
+ 16:41:36,160 INFO  MyThread:17 - 38
+ 16:41:36,160 INFO  MyThread:17 - 39
+ 16:41:36,160 INFO  MyThread:17 - 40
+
```
Comment (YuraBondarenko) [inline log/logger.log:41]: add empty line

**Review by yurwing — APPROVED** [review]

File: log/logger.log (hunk 1 above)
Comment (RomanDubovskyi) [inline log/logger.log]: missing empty line at the end of the file

File: src/main/java/com/dev/Main.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.dev;
+
+public class Main {
+    public static void main(String[] args) {
+        Counter counter = new Counter();
+        counter.setCounter(0);
+        MyThread myThread = new MyThread(counter);
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/Main.java]:
```suggestion
        Counter counter = new Counter();
        MyThread myThread = new MyThread(counter);
```
it's primitive it'll be 0 by default

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/Main.java (hunk 2 above)
Comment (IgorBochkor) [reply src/main/java/com/dev/Main.java]: of course, fixed)

File: log/logger.log (hunk 1 above)
Comment (IgorBochkor) [reply log/logger.log]: fixed

File: src/main/java/com/dev/MyThread.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    private static final Logger logger = Logger.getLogger(MyThread.class);
+    private Counter counter;
+
+    public MyThread(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        for (int i = 0; i < 20; i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/MyThread.java:15]:
```suggestion
       while (counter.getCounter() < 20) {
```
you shouldn't do it 20 times, but until the counter will reach a certain value

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**